src/services/socket.tsx

```diff
@@ -3,40 +3,87 @@
 import { io } from 'socket.io-client'
 import { useEffect, useState } from 'react'
 
-type ListenerRes = {
-	messageServer: string
-	roomIdServer: string
-	userNameServer: string
+export type ListenerRes = {
+	message: string
+	// roomId: string
+	userName: string
+	timeStamp: string
 }
 
-export interface Event extends ListenerRes {
+type AllUsersPointsRes = {
+	message: ListenerRes[]
+	// roomId: string
+	userName: string
 	timeStamp: string
 }
 
 const socket = io()
 
 export function socketRoomEmitter(
 	emitterName: string,
-	message: string,
-	roomId: string,
+	message: string | ListenerRes[],
 	userName: string,
+	timeStamp: string,
+	roomId: string,
 ) {
-	socket.emit(emitterName, message, roomId, userName)
+	socket.emit(emitterName, message, userName, timeStamp, roomId)
+}
+
+export function useAllUsersPointsListener(listenerName: string) {
+	const [listenerRes, setListenerRes] = useState<AllUsersPointsRes>()
+	useEffect(() => {
+		function onListenerRes(
+			message: ListenerRes[],
+			// roomId: string,
+			userName: string,
+			timeStamp: string,
+		) {
+			setListenerRes({ message, userName, timeStamp })
+		}
+		socket.on(listenerName, onListenerRes)
+
+		return () => {
+			socket.off(listenerName, onListenerRes)
+		}
+	}, [listenerName])
+	return listenerRes
 }
 
 export function useSocketListener(listenerName: string) {
-	const [listenerRes, setListenerRes] = useState<ListenerRes>({
-		messageServer: '',
-		roomIdServer: '',
-		userNameServer: '',
-	})
+	const [listenerRes, setListenerRes] = useState<ListenerRes>()
+
 	useEffect(() => {
 		function onListenerRes(
-			messageServer: string,
-			roomIdServer: string,
-			userNameServer: string,
+			message: string,
+			// roomId: string,
+			userName: string,
+			timeStamp: string,
 		) {
-			setListenerRes({ messageServer, roomIdServer, userNameServer })
+			setListenerRes({ message, userName, timeStamp })
+		}
+		socket.on(listenerName, onListenerRes)
+
+		return () => {
+			socket.off(listenerName, onListenerRes)
+		}
+	}, [listenerName])
+
+	return listenerRes
+}
+
+export function useSocketListenerWithCb(listenerName: string, config: any) {
```

Comment:
todo: Why did you create a new function? The new param (config) could added to `useSocketListener` as an optional.

```js
type Config = {
  callback: (value: ListenerRes) => void
}

export function useSocketListener(eventName: string, config?: Config) {
  // ...
}

```

Reply:
I had created a 2nd one with a callback. I've modified this to be one with an optional callback function. It'll be in the next commit.